includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1485,19 +1486,39 @@ public function process_refund( $order_id, $amount = null, $reason = '' ) {
 
 		if ( empty( $reason ) ) {
 			$note = sprintf(
-				/* translators: %1: the successfully charged amount */
-				__( 'A refund of %1$s was successfully processed using WooCommerce Payments.', 'woocommerce-payments' ),
-				WC_Payments_Explicit_Price_Formatter::get_explicit_price( wc_price( $amount, [ 'currency' => $currency ] ), $order )
+				WC_Payments_Utils::esc_interpolated_html(
+					/* translators: %1: the successfully charged amount, %2: refund id */
+					__( 'A refund of %1$s was successfully processed using WooCommerce Payments (<code>%2$s</code>).', 'woocommerce-payments' ),
+					[
+						'code' => '<code>',
+					]
+				),
+				WC_Payments_Explicit_Price_Formatter::get_explicit_price( wc_price( $amount, [ 'currency' => $currency ] ), $order ),
+				$refund['id']
 			);
 		} else {
 			$note = sprintf(
-				/* translators: %1: the successfully charged amount, %2: reason */
-				__( 'A refund of %1$s was successfully processed using WooCommerce Payments. Reason: %2$s', 'woocommerce-payments' ),
+				WC_Payments_Utils::esc_interpolated_html(
+					/* translators: %1: the successfully charged amount, %2: refund id, %3: reason */
+					__( 'A refund of %1$s was successfully processed using WooCommerce Payments (<code>%2$s</code>). Reason: %3$s', 'woocommerce-payments' ),
+					[
+						'code' => '<code>',
+					]
+				),
 				WC_Payments_Explicit_Price_Formatter::get_explicit_price( wc_price( $amount, [ 'currency' => $currency ] ), $order ),
+				$refund['id'],
 				$reason
 			);
 		}
 
+		// Get the last created WC refund from order and save WCPay refund id as meta.
+		$wcpay_db       = new WC_Payments_DB();
+		$wc_last_refund = $wcpay_db->last_refund_from_order_id( $order_id );
```

Comment:
I think it would be better to move the util method `last_refund_from_order_id` to a public static method in `WC_Payments_Utils` due to 2 reasons: 

- Initializing this service `WC_Payments_DB` is a bit contrast with our effort of currently using it and plug into it the relevant services in the entry class `WC_Payments`. 
- The method itself is not tightly coupled with WCPay, and it can be used by anywhere else in our codebase. Hence, a public static function `WC_Payments_Utils` is a good fit.